### PR TITLE
Chromatic (storybook 6.4)

### PR DESCRIPTION
**Pull request by orrgottlieb**
## Go over this checklist before submitting your Pull Request

Description: **ADD YOUR PR DESCRIPTION HERE**

STEP 1: Fulfill the description of the PR above

STEP 2: Select one of 2 main scenarios of adding the new component or editing the existing one

STEP 3: Delete checklist which you don't need

STEP 4: Go over the checklist. All checkboxes should be marked

STEP 5: Delete this instruction

STEP 6: Happy code review and merge!

#### Basic
- [ ] Used plop (npm run plop) to create a new component. This command will create a folder for your component and will update index.js file
- [ ] PR has description
- [ ] New component is functional and using React Hooks. 
- [ ] New React Hooks were added to /src/hooks folder (optional)
- [ ] Component defines PropTypes
- [ ] There is Readme for the component, which explains the main aspects of how to use it
#### Style
- [ ] Styles are added to NewComponent.scss file inside of the NewComponent folder
- [ ] CSS selectors are named using [BEM convention](http://getbem.com/naming/) 
- [ ] Design is compatible with [Monday Design System](https://design.monday.com/)
#### Storybook
- [ ] Stories were added to /src/NewComponent/__stories__/NewComponent.stories.js file
- [ ] Stories include sandbox story with knobs for each property
- [ ] Stories include all flows of using the component
- [ ] Component passed Accessibility Plugin checks
#### Tests
- [ ] Tests are compliant with TESTING_README.md instructions

#### Basic
- [ ] PR has description
- [ ] Changes to the component are backward compatible (including selectors structure). If not - add to the title of the PR "BREAKABLE_CHANGE""
- [ ] All changes to the component are reflected in the ReadMe
- [ ] If component is old and was not compliant with the latest guidelines - it was fixed (optional) 
#### Style
- [ ] CSS selectors are named using [BEM convention](http://getbem.com/naming/) 
- [ ] Design is compatible with [Monday Design System](https://design.monday.com/)
#### Storybook
- [ ] All the changes to the component should be reflected in the Storybook.
- [ ] Component passed Accessibility Plugin checks
#### Tests
- [ ] All current tests are passing
- [ ] New functionality is covered with tests
- [ ] Tests are compliant with TESTING_README.md instructions

### CODE REVIEW

File: .storybook/preview.js [hunk 1]
```diff
@@ -13,10 +13,12 @@ import {
   DocFooter,
   MultipleStoryElementsWrapper
 } from "../src/storybook/components";
+import LinkComponent from "../src/storybook/components/link-component/link-component";
 
 addParameters({
   docs: {
     ...parameters.docs,
+    inlineStories: true,
```
Comment (hadasfa) [inline .storybook/preview.js:21]: this for stop put the stories in iframe right?

File: plop/hooks/hook-story.txt [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import { useRef, useCallback } from "react"
+import { ArgsTable, Story, Canvas, Meta } from "@storybook/addon-docs";
```
Comment (hadasfa) [inline plop/hooks/hook-story.txt:2]: cool!

File: plop/hooks/index.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    ],
+    actions: [
+      {
+        type: "add",
+        path: "src/hooks/__stories__/{{camelCase componentName}}.stories.mdx",
+        templateFile: "plop/hooks/hook-story.txt"
+      },
+      {
+        type: "add",
+        path: "src/hooksDummyComponents/{{properCase componentName}}.jsx",
```
Comment (hadasfa) [inline plop/hooks/index.js:19]: I kind of not understand this flow. Why we create a dummy component for a new hook?  Is this only fot the meta and props component in the story?

File: src/components/Button/__stories__/button.stories.mdx [hunk 4]
```diff
@@ -228,4 +228,4 @@ When one action is selected, the other is disabled
 </Canvas>
 
 ## Related components
-<RelatedComponents componentsNames={[SPLIT_BUTTON, BUTTON_GROUP, BUTTON_GROUP]}/>
\ No newline at end of file
+<RelatedComponents componentsNames={[SPLIT_BUTTON, BUTTON_GROUP, BUTTON_GROUP]}/>
```
Comment (hadasfa) [inline src/components/Button/__stories__/button.stories.mdx:231]: what is the difference here?

File: src/components/IconButton/__stories__/IconButton.stories.mdx [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+- [Use cases and examples](#use-cases-and-examples)
+- [Related components](#related-components)
+- [Feedback](#feedback)
+
+## Overview
+Please write here the component description
+
+<Canvas>
+  <Story name="Overview" args={{}}>
+    <IconButton ariaLabel="Add Button"/>
```
Comment (hadasfa) [inline src/components/IconButton/__stories__/IconButton.stories.mdx:36]: We need to use the iconButtonTemplate for the overview story for creating access to the controls in the sandbox (like in the button story)

File: src/hooks/__stories__/useClickOutside.stories.mdx [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+this hook is used when you want to capture click events outside of your component
+
+<Canvas>
+  <Story name="Overview">
+    {() => {
+      const [counter, setCounter] = useState(0);
+      const ref = useRef(null);
+      const onClick = useCallback(() => {
+       setCounter(currentCounter => {debugger; return currentCounter + 1})
```
Comment (hadasfa) [inline src/hooks/__stories__/useClickOutside.stories.mdx:26]: I think we can remove the debugger;

File: src/storybook/components/tip/tip.jsx [hunk 7]
```diff
@@ -9,7 +9,11 @@ export const Tip = ({ className, title, children }) => {
       icon={null}
       type={CoreAttentionBox.types.DARK}
       componentClassName={cx("monday-storybook-tip", className)}
-      title={`🤓 ${title}`}
+      title={
+        <span>
```
Comment (hadasfa) [inline src/storybook/components/tip/tip.jsx:13]: fix linter error

File: src/storybook/components/usage-guidelines/usage-guideline.scss [hunk 8]
```diff
@@ -4,16 +4,23 @@
 .monday-storybook-usage-guidelines{
   margin-top: $spacing-between-section-items;
   &_guideline {
-    margin: $spacing-medium 0;
+    color: var(--primary-text-color);
+    margin: var(--spacing-medium) var(--spacing-small);
     display: flex;
     align-items: center;
+    font-size: 16px;
```
Comment (hadasfa) [inline src/storybook/components/usage-guidelines/usage-guideline.scss:11]: we usually using line-height: 24px in other text components. is the guideline should be different?
also there are mixins for text in the storybook new design (storybook/styles/typography) - but there we using 24 and not 21 line height.

File: src/storybook/components/usage-guidelines/usage-guideline.scss [hunk 9]
```diff
@@ -4,16 +4,23 @@
...
+    line-height: 21px;
     font-weight: 300;
     &:first-child {
       margin-top: 0;
     }
   }
 
   &_icon {
-    margin-right: $spacing-small;
+    font-size: 16px;
```
Comment (hadasfa) [inline src/storybook/components/usage-guidelines/usage-guideline.scss:20]: same note about the line height. is this intentional?

File: .storybook/preview-head.html [hunk 10]
```diff
@@ -31,15 +28,66 @@
...
   }
   .black-app-theme {
-      background-color: var(--primary-background-color);
+    background-color: var(--primary-background-color);
   }
   .hacker_theme-app-theme {
-      background-color: var(--primary-background-color);
+    background-color: var(--primary-background-color);
   }
+  .sbdocs.sbdocs-wrapper,
```
Comment (hadasfa) [inline .storybook/preview-head.html:42]: These changes is for the themes support right?

File: src/hooks/__stories__/useClickOutside.stories.mdx (hunk 6 above)
Comment (orrgottlieb) [reply src/hooks/__stories__/useClickOutside.stories.mdx:26]: thanks, will remove

File: src/components/IconButton/__stories__/IconButton.stories.mdx (hunk 5 above)
Comment (orrgottlieb) [reply src/components/IconButton/__stories__/IconButton.stories.mdx:36]: will change

File: src/components/Button/__stories__/button.stories.mdx (hunk 4 above)
Comment (orrgottlieb) [reply src/components/Button/__stories__/button.stories.mdx:231]: good questions

File: plop/hooks/index.js (hunk 3 above)
Comment (orrgottlieb) [reply plop/hooks/index.js:19]: the ArgsTable can accept different settings, but unless it receives a component.
So in order for an easier way to populate the hook's arguments, I created a dummy component which we will edit and config the arguments

File: .storybook/preview.js (hunk 1 above)
Comment (orrgottlieb) [reply .storybook/preview.js:21]: should :)

File: src/storybook/components/usage-guidelines/usage-guideline.scss (hunk 9 above)
Comment (orrgottlieb) [reply src/storybook/components/usage-guidelines/usage-guideline.scss:20]: the font was a bit bigger (it looked odd)

File: src/components/IconButton/__stories__/IconButton.stories.mdx (hunk 5 above)
Comment (orrgottlieb) [reply src/components/IconButton/__stories__/IconButton.stories.mdx:36]: current issue with storybook 6.4

File: src/components/IconButton/__tests__/iconButton-snapshot-tests.jest.js [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+    const tree = renderer.create(<IconButton />).toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+
+    it("with id", () => {
+        const tree = renderer.create(<IconButton id="icon-button" />).toJSON();
+        expect(tree).toMatchSnapshot();
+    });
+
+    it("with", () => {
```
Comment (hadasfa) [inline src/components/IconButton/__tests__/iconButton-snapshot-tests.jest.js:47]: with class name?

File: src/components/IconButton/__tests__/iconButton-snapshot-tests.jest.js [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+        const tree = renderer.create(<IconButton className="class-name" />).toJSON();
+        expect(tree).toMatchSnapshot();
+    });
+
+  it("with disabled", () => {
+    const tree = renderer.create(<IconButton disabled />).toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+
+    //   it(`with size - xs`, () => {
```
Comment (hadasfa) [inline src/components/IconButton/__tests__/iconButton-snapshot-tests.jest.js:57]: if note does not needed can we remove it?

File: src/components/IconButton/__stories__/IconButton.stories.mdx (hunk 5 above)
Comment (hadasfa) [reply src/components/IconButton/__stories__/IconButton.stories.mdx:36]: what do you mean?

**Pull request merged**